gum/backend-arm64/guminterceptor-arm64.c

```diff
@@ -1147,3 +1148,29 @@ gum_emit_epilog (GumArm64Writer * aw)
   gum_arm64_writer_put_pop_reg_reg (aw, ARM64_REG_X16, ARM64_REG_X17);
   gum_arm64_writer_put_br_reg (aw, ARM64_REG_X16);
 }
+
+static void
+gum_emit_epilog_leave (GumArm64Writer * aw)
+{
+  guint i;
+
+  /* Skip PC and SP */
+  gum_arm64_writer_put_add_reg_reg_imm (aw, ARM64_REG_SP, ARM64_REG_SP, 16);
+
+  /* Restore NZCV and X0 */
+  gum_arm64_writer_put_pop_reg_reg (aw, ARM64_REG_X1, ARM64_REG_X0);
+  gum_arm64_writer_put_mov_nzcv_reg (aw, ARM64_REG_X1);
+
+  /* Restore X1-X28, FP, and LR */
+  for (i = 1; i != 29; i += 2)
+    gum_arm64_writer_put_pop_reg_reg (aw, ARM64_REG_X0 + i, ARM64_REG_X1 + i);
+  gum_arm64_writer_put_pop_reg_reg (aw, ARM64_REG_FP, ARM64_REG_LR);
+
+  /* Restore vector registers */
+  for (i = 0; i != 32; i += 2)
+    gum_arm64_writer_put_pop_reg_reg (aw, ARM64_REG_Q0 + i, ARM64_REG_Q1 + i);
+
+  gum_arm64_writer_put_pop_reg_reg (aw, ARM64_REG_X16, ARM64_REG_X17);
+  /* ret X16 to fix BTI issue */
+  gum_arm64_writer_put_instruction (aw, 0xd65f0200);
```

Comment:
Could we do this for both epilogues instead of special-casing?

Reply:
sorry for late reply, please review the latest commit